soroban-spec/src/gen/ts/project_template/src/invoke.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  isConnected,
+  getPublicKey,
+  signTransaction,
+} from "@stellar/freighter-api";
+import * as SorobanClient from 'soroban-client'
+import { Buffer } from "buffer";
+import type { Account, Memo, MemoType, Operation, Transaction } from 'soroban-client';
+import { NETWORK_PASSPHRASE, CONTRACT_ID } from './constants'
+import { Server } from './server'
+
+
+export type Tx = Transaction<Memo<MemoType>, Operation[]>
+
+export type Simulation = NonNullable<SorobanClient.SorobanRpc.SimulateTransactionResponse['results']>[0]
+
+export type TxResponse = SorobanClient.SorobanRpc.GetTransactionResponse;
+
+export type InvokeArgs = {
+  method: string
+  args?: any[]
+  sign?: boolean
+  fee?: number
+}
+
+/**
+ * Get account details from the Soroban network for the publicKey currently
+ * selected in Freighter. If not connected to Freighter, throws errors. Will
+ * pop up Freighter's modal to request user permissions, if this hasn't been
+ * done already.
+ */
+export async function getAccount(): Promise<Account> {
+  if (!await isConnected()) {
+    throw new Error('Freighter not connected')
+  }
+  const publicKey = await getPublicKey()
+  if (!publicKey) {
+    throw new Error('Freighter not initialized')
+  }
+  return await Server.getAccount(publicKey)
+}
+
+export class NotImplementedError extends Error { }
+
+/**
+ * Invoke a method on the INSERT_CONTRACT_NAME_HERE contract.
+ *
+ * Uses Freighter to determine the current user and sign the transaction.
+ *
+ * @param {string} obj.method - The method to invoke.
+ * @param {any[]} obj.args - The arguments to pass to the method.
+ * @param {boolean} obj.sign - Whether to sign the transaction with Freighter.
```

Comment:
Since this method is just doing a `simulateTransaction` it doesn't actually need to sign anything. And this arg seems unused anyway.

It would be nice if the method name and/or docs indicated it was not going to actually send a real live transaction to the network.

Reply:
Wait.... it sometimes sends the txn? what?

Reply:
Yeah we need to fix it to check the footprint instead of the auth.

Reply:
Oh, is this the thing where you're trying to auto-infer if it is a read call or a write call? That will never work correctly. I'd strongly argue for a more viem style [read](https://viem.sh/docs/ethers-migration.html#reading-from-contracts)/[write](https://viem.sh/docs/ethers-migration.html#writing-to-contracts) interface where the dev specifically says which they want to do.